### DISCUSSION TITLE
Multi-tenancy per workspace

### DISCUSSION THREAD

**Discussion by TheTinkerBasement** [tryethernal/ethernal#245] [Feature Requests]
Hi,
I have some friends who would like to access my workspace but I don't see a way how this is possible at the moment.

There is a 1:1 relationship between Workspace and Hardhat blockchain instance (through the Api-key)
and a 
1:1 relationship between User accounts and this workspace.

We would need a n:1 relationship between User accounts and workspace though.

Best,
Comedian

**Comment by antoinedc** [comment]
At the moment, the only way to do that is to setup a public explorer. With the self-hosted version, you can use this script:
https://doc.tryethernal.com/getting-started/self-hosted#setup-a-public-explorer
You should just need to replace constants at the beginning of it, and then run it. Let me know if you have issues with it.

Basically, this will make workspace data public through a URL, and let anyone have a read-only access (+ contract interaction) to your workspace.
Actions such as resetting the workspace, deleting a contract, etc, will still only be available when logged in.

Does that solve your issue?

**Reply by TheTinkerBasement** [reply]
yes, does solve this issue! Thank you

**Reply by antoinedc** [reply]
By the way, do you mind pinging me on Discord? I don't know if you joined it. I don't think there are many production deployments of Ethernal, so I'd be interested in keeping in touch to see how it goes :)
